base64.go

```diff
@@ -26,13 +26,31 @@ import (
 // When the bytes are unmarshalled from JSON they are decoded from base64.
 type Base64String []byte
 
+// Encode encodes the bytes as base64
+func (b64 Base64String) Encode() string {
+	return base64.RawStdEncoding.EncodeToString(b64)
+}
+
+// Decode decodes the given input into this Base64String
+func (b64 *Base64String) Decode(str string) error {
+	// We must check whether the string was encoded in a URL-safe way in order
+	// to use the appropriate encoding.
+	var err error
+	if strings.ContainsAny(str, "-_") {
```

Comment:
*screams*

Why is this required? Don't we specify the base64 encoding? Is this to avoid the need for two different `Base64String` types? Are you just keeping it the way you found it because who knows what will break otherwise?

Reply:
Well, I was deliberately trying to avoid changing too much at once, so yes I was just preserving the original code.

It looks like @NegativeMjark and @babolivier have already been down this path: https://github.com/matrix-org/gomatrixserverlib/pull/58#pullrequestreview-61246401

Reply:
Fair, though ewwwwwwwwwww